R/request.R

```diff
@@ -122,6 +122,12 @@ request_prepare <- function(req) {
 
 request_perform <- function(req, handle, refresh = TRUE) {
   stopifnot(is.request(req), inherits(handle, "curl_handle"))
+
+  orig_req <- req
```

Comment:
Can you this and the block below up into a small helper function that lives in `callback.r`?